main.go

```diff
@@ -188,12 +188,14 @@ func main() {
 
 	decoder := admission.NewDecoder(mgr.GetScheme())
 
-	if os.Getenv("DISABLE_AUTO_ANNOTATION") != "true" {
+	if os.Getenv("DISABLE_AUTO_ANNOTATION") == "true" || autoAnnotationConfigStr == "" {
```

Comment:
nit: Would rather use something like https://github.com/caarlos0/env, so that we can parse the env variables to proper types.

Reply:
Would rather not pull in this dependency.